### PR TITLE
[G1-5584] update snap agent docker file

### PR DESCRIPTION
**Pull request by shanas-swi**
update yq to latest version, which has vulnerability fixes. init.sh file yq commands also modified. 
in docker file, docker.io removed due to vulnerability and introduced official docker

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,8 +14,12 @@ RUN apt-get update && \
       apt-transport-https \
       ca-certificates \
       curl \
-      docker.io \
```
Comment (shanas-swi) [inline Dockerfile:17]: docker.io removed due to vulnerabilities

File: Dockerfile [hunk 2]
```diff
@@ -14,8 +14,12 @@ RUN apt-get update && \
...
-      gnupg && \
+      gnupg \
+      lsb-release && \
+      curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg && \
```
Comment (shanas-swi) [inline Dockerfile:19]: included official docker instead of ubuntu docker.io

File: Dockerfile [hunk 3]
```diff
@@ -24,7 +28,7 @@ RUN arch="$(uname -m)" && if [ "${arch}" = "aarch64" ]; then \
     elif [ "${arch}" = "x86_64" ]; then \
       yq_arch=amd64; \ 
     fi && \ 
-    curl -L "https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_${yq_arch}" -o yq && \
+    curl -L "https://github.com/mikefarah/yq/releases/download/v4.44.3/yq_linux_${yq_arch}" -o yq && \
```
Comment (shanas-swi) [inline Dockerfile:31]: added latest version of yq due to vulnerabilities. we need to use correct yq command format to update yaml files.

File: Dockerfile [hunk 4]
```diff
@@ -34,6 +38,8 @@ RUN echo "deb https://packagecloud.io/solarwinds/${swisnap_repo}/ubuntu/ focal m
   apt-get update && \
   apt-get -y install solarwinds-snap-agent=${swisnap_version} && \
   usermod -aG root solarwinds && \
+  getent group docker || groupadd docker && \
```
Comment (shanas-swi) [inline Dockerfile:41]: docker requires these permission

File: conf/swisnap-init.sh [hunk 5]
```diff
@@ -27,29 +27,29 @@ swisnap_config_setup() {
         exit 1
     fi
 
-    yq w -i "${PUBLISHER_APPOPTICS_CONFIG}" v1.publisher.publisher-appoptics.all.token -- "${SWI_TOKEN}"
-    yq w -i "${PUBLISHER_APPOPTICS_CONFIG}" v2.publisher.publisher-appoptics.all.endpoint.token -- "${SWI_TOKEN}"
-    yq w -i "${PUBLISHER_PROCESSES_CONFIG}" v2.publisher.publisher-processes.all.endpoint.token -- "${SWI_TOKEN}"
+    yq eval -i '.v1.publisher."publisher-appoptics".all.token = "'"${SWI_TOKEN}"'"' "${PUBLISHER_APPOPTICS_CONFIG}"
+    yq eval -i '.v2.publisher."publisher-appoptics".all.endpoint.token = "'"${SWI_TOKEN}"'"' "${PUBLISHER_APPOPTICS_CONFIG}"
```
Comment (shanas-swi) [inline conf/swisnap-init.sh:31]: we could see this is updating the token

publisher-v2

File: conf/swisnap-init.sh [hunk 6]
```diff
@@ -27,29 +27,29 @@ swisnap_config_setup() {
         exit 1
     fi
 
-    yq w -i "${PUBLISHER_APPOPTICS_CONFIG}" v1.publisher.publisher-appoptics.all.token -- "${SWI_TOKEN}"
-    yq w -i "${PUBLISHER_APPOPTICS_CONFIG}" v2.publisher.publisher-appoptics.all.endpoint.token -- "${SWI_TOKEN}"
-    yq w -i "${PUBLISHER_PROCESSES_CONFIG}" v2.publisher.publisher-processes.all.endpoint.token -- "${SWI_TOKEN}"
+    yq eval -i '.v1.publisher."publisher-appoptics".all.token = "'"${SWI_TOKEN}"'"' "${PUBLISHER_APPOPTICS_CONFIG}"
+    yq eval -i '.v2.publisher."publisher-appoptics".all.endpoint.token = "'"${SWI_TOKEN}"'"' "${PUBLISHER_APPOPTICS_CONFIG}"    
+    yq eval -i '.v2.publisher."publisher-processes".all.endpoint.token = "'"${SWI_TOKEN}"'"' "${PUBLISHER_PROCESSES_CONFIG}"
```
Comment (shanas-swi) [inline conf/swisnap-init.sh:32]: we could see this is updating the processor token

processor-v2

File: conf/swisnap-init.sh [hunk 7]
```diff
@@ -160,10 +155,8 @@ run_plugins_with_default_configs() {
         KUBERNETES_LOGS_CONFIG="${TASK_AUTOLOAD_DIR}/task-logs-k8s-events.yaml"
         if check_if_plugin_supported "Kubernetes Logs" "${KUBERNETES_LOGS_CONFIG}.example"; then
             mv "${KUBERNETES_LOGS_CONFIG}.example" "${KUBERNETES_LOGS_CONFIG}"
-            yq w -i "${KUBERNETES_LOGS_CONFIG}" 'plugins.(plugin_name==k8s-events).config.incluster' 'true'
-            yq w -i "${KUBERNETES_LOGS_CONFIG}" 'plugins.(plugin_name==k8s-events).config.filters[+].namespace' 'default'
-            yq w -i "${KUBERNETES_LOGS_CONFIG}" 'plugins.(plugin_name==k8s-events).config.filters[*].watch_only' 'true'
-            yq w -i "${KUBERNETES_LOGS_CONFIG}" 'plugins.(plugin_name==k8s-events).config.filters[*].options.fieldSelector' 'type==Normal'
+            yq eval -i '.plugins[] |= select(.plugin_name == "k8s-events").config.incluster = true' "${KUBERNETES_LOGS_CONFIG}"
+            yq eval -i '.plugins[] |= select(.plugin_name == "k8s-events").config.filters = [{"namespace": "default", "watch_only": true, "options": {"fieldSelector": "type==Normal"}}]' "${KUBERNETES_LOGS_CONFIG}"
```
Comment (shanas-swi) [inline conf/swisnap-init.sh:159]: we could see this updates k8s-events

k8s-events

**Review by mpach-swi — APPROVED** [review]

**Comment by mpach-swi** [comment]
@shanas-swi remember about commit squash

**Pull request merged**